cpu/nrf51/include/periph_cpu.h

```diff
@@ -26,6 +26,18 @@ extern "C" {
 #endif
 
 /**
+ * @brief   Mandatory macro for defining GPIO pins
+ *
+ * The port definition is used (and zeroed) to suppress compiler warnings
+ */
+#define GPIO_PIN(x,y)       ((x & 0) | y)
```

Comment:
There might be some minor unrelated (though the commit states that some cleanup is included...), but it would take me days to make every single fix into it's own commit/PR...